src/utils/fs.ts

```diff
@@ -41,34 +41,34 @@ export const safeUnlink = async (path: string) => {
 // filenames within those directories, if at least one of the directories
 // exists. If not, an error is thrown.
 export const listFunctionsDirectories = async function (srcFolders: string[]) {
-  const filenamesByDirectory = await Promise.all(
-    srcFolders.map(async (srcFolder) => {
-      try {
-        const filenames = await listFunctionsDirectory(srcFolder)
+  const filenamesByDirectory = await Promise.allSettled(
+    srcFolders.map((srcFolder) => listFunctionsDirectory(srcFolder)),
+  )
+  const errorMessages: string[] = []
+  const validDirectories = filenamesByDirectory
+    .map((result) => {
+      if (result.status === 'rejected') {
+        errorMessages.push(result.reason)
 
-        return filenames
-      } catch {
         return null
       }
-    }),
-  )
-  const validDirectories = filenamesByDirectory.filter(nonNullable)
+
+      return result.value
+    })
+    .filter(nonNullable)
 
   if (validDirectories.length === 0) {
-    throw new Error(`Functions folder does not exist: ${srcFolders.join(', ')}`)
+    throw new Error(`Functions folders do not exist: ${srcFolders.join(', ')}
+${errorMessages.join('\n')}`)
```

Comment:
I understand that you're doing this to capture additional information about the error for debugging purposes, but I worry this will add noise and confusion to an error message that was short and to the point. Any thoughts on how we could access this information without adding noise to the actual message we present to users?

Reply:
But the error message might also not be accurate. If for whatever reason there is an IO error, we still show this message, even though It might be a different error. (I think this is what actually happened in https://github.com/netlify/pod-compute/issues/108, because in the build logs it first prints the functions it found in these folders and then an error that the functions folders do not exist.)

But to your point: We could introduce a logLevel, that is by default different in build and cli.

Reply:
Other solution would be to check if the error is `fileNotFound` for our nice error and otherwise let the error bubble.

Reply:
Done